src/common/options/global.yaml.in

```diff
@@ -3083,7 +3083,7 @@ options:
     OSD refuses to create new PGs. An OSD stops creating new PGs if the number
     of PGs it serves exceeds
     ``osd_max_pg_per_osd_hard_ratio`` \* ``mon_max_pg_per_osd``.
-  default: 3
+  default: 10
```

Comment:
At first glance it looks like a treatment of a symptom.

```cpp
PGRef OSD::handle_pg_create_info(const OSDMapRef& osdmap,
                                 const PGCreateInfo *info)
{
  spg_t pgid = info->pgid;

  if (maybe_wait_for_max_pg(osdmap, pgid, info->by_mon)) {
    dout(10) << __func__ << " hit max pg, dropping" << dendl;
    return nullptr;
  }
  // ...
```

```cpp
bool OSD::maybe_wait_for_max_pg(const OSDMapRef& osdmap,
                                spg_t pgid,
                                bool is_mon_create)
{
  const auto max_pgs_per_osd =
    (cct->_conf.get_val<uint64_t>("mon_max_pg_per_osd") *
     cct->_conf.get_val<double>("osd_max_pg_per_osd_hard_ratio"));

  if (num_pgs < max_pgs_per_osd) {
    return false;
  }

  // ...
  dout(1) << __func__ << " withhold creation of pg " << pgid
          << ": " << num_pgs << " >= "<< max_pgs_per_osd << dendl;
  return true;
}
```

I wonder why there was a storm of PG creation events. I recall some reports on misbehavior of the PG autoscaler (CC @pdvian) resulting in very aggressive splitting.

Reply:
I've experienced this issue personally a few times. In most cases it happened when adding a whole new empty host.